src/Rules/FunctionDefinitionCheck.php

```diff
@@ -26,6 +26,7 @@ class FunctionDefinitionCheck
 		'float',
 		'void',
 		'iterable',
+		'object',
```

Comment:
This does not scale, it better to change to constant to be a map of typehint => minimal PHP version, i.e.

~~~php
const VALID_TYPEHINTS = [
	'self' => 50000,
	'parent' => 50000,
	'array' => 50000,
	'callable' => 50400,
	'string' => 70000,
	'int' => 70000,
	'bool' => 70000,
	'float' => 70000,
	'void' => 70100,
	'iterable' => 70100,
	'object' => 70200,
];
~~~

Reply:
Note that I also dropped `static` and added `parent`